satpy/readers/utils.py

```diff
@@ -210,57 +233,84 @@ def unzip_file(filename, prefix=None):
         Temporary filename path for decompressed file or None.
 
     """
-    if os.fspath(filename).endswith('bz2'):
-        fdn, tmpfilepath = tempfile.mkstemp(prefix=prefix,
-                                            dir=config["tmp_dir"])
-        LOGGER.info("Using temp file for BZ2 decompression: %s", tmpfilepath)
-        # try pbzip2
-        pbzip = which('pbzip2')
-        # Run external pbzip2
-        if pbzip is not None:
-            n_thr = os.environ.get('OMP_NUM_THREADS')
-            if n_thr:
-                runner = [pbzip,
-                          '-dc',
-                          '-p'+str(n_thr),
-                          filename]
-            else:
-                runner = [pbzip,
-                          '-dc',
-                          filename]
-            p = Popen(runner, stdout=PIPE, stderr=PIPE)  # nosec
-            stdout = BytesIO(p.communicate()[0])
-            status = p.returncode
-            if status != 0:
-                raise IOError("pbzip2 error '%s', failed, status=%d"
-                              % (filename, status))
-            with closing(os.fdopen(fdn, 'wb')) as ofpt:
-                try:
-                    stdout.seek(0)
-                    shutil.copyfileobj(stdout, ofpt)
-                except IOError:
-                    import traceback
-                    traceback.print_exc()
-                    LOGGER.info("Failed to read bzipped file %s",
-                                str(filename))
-                    os.remove(tmpfilepath)
-                    raise
-            return tmpfilepath
-
-        # Otherwise, fall back to the original method
-        bz2file = bz2.BZ2File(filename)
-        with closing(os.fdopen(fdn, 'wb')) as ofpt:
-            try:
-                ofpt.write(bz2file.read())
-            except IOError:
-                import traceback
-                traceback.print_exc()
-                LOGGER.info("Failed to read bzipped file %s", str(filename))
-                os.remove(tmpfilepath)
-                return None
-        return tmpfilepath
-
-    return None
+    fdn, tmpfilepath = tempfile.mkstemp(prefix=prefix,
+                                        dir=config["tmp_dir"])
+    LOGGER.info("Using temp file for BZ2 decompression: %s", tmpfilepath)
+    # check pbzip2 status
+    pbzip2 = _unzip_with_pbzip(filename, tmpfilepath, fdn)
+    if pbzip2 is not None:
+        return pbzip2
+    # Otherwise, fall back to the original method
+    bz2file = bz2.BZ2File(filename)
+    content = bz2file.read()
```

Comment:
For better readability, this could be extracted to a separate method `_unzip_with_bz2file`

Reply:
OK, I have extracted this with `_unzip_with_bz2file` and added IOerror catch for file reading.